src/components/ComponentSettingsModal.jsx

```diff
@@ -281,21 +307,26 @@ export default class ComponentSettingsModal extends Component {
               <div className={elementStyles.modalFormFields}>
                 {/* Category */}
                 <p className={elementStyles.formLabel}>Category Folder Name</p>
-                <div className="d-flex">
-                  <select className="w-100" id="category" value={category} onChange={this.changeHandler} disabled={isCategoryDropdownDisabled(isNewFile, prevCategory)} >
-                  {
-                    allCategories
+                <div className="d-flex text-nowrap">
+                  <CreatableSelect
+                    isClearable
+                    className="w-100"
+                    onChange={this.selectChangeHandler}
+                    isDisabled={isCategoryDropdownDisabled(isNewFile, prevCategory)}
```

Comment:
Currently, since the `isDisabled` prop is dependent on `prevCategory`, which is a calculated state variable, when we're in a collection, the dropdown is actually temporarily active before being disabled. This results in bad UX if a user eagerly clicks on it and manages to type something before it gets disabled.

Since the `prevCategory` is actually the original `category` prop, what we can do instead is to pass an `isCatgeoryDropdownDisabled` prop from `CollectionPagesSection` to `ComponentSettingsModal`, which takes the value of `isCategoryDropdownDisabled(createNewPage, collectionName)` (moving the `isCategoryDropdownDisabled helper function to the `CollectionPagesSection` component). This way the dropdown will be disabled from the start when we're in a category.

We should also set the default value based on the `category` prop instead of the `category` state, because `category` state represents the new category values, which results in `"Select a category or create a new category..."` being displayed inside the `CategoryPage` layout.

Reply:
This might be a bit confusing, so lmk if you need clarification

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/207/commits/b24c6bca57ad078e586e66c5c79013b858a5071f